pkg/apis/ark/v1/snapshot.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2018 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Snapshot is... TODO
+type Snapshot struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   SnapshotSpec   `json:"spec"`
+	Status SnapshotStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// SnapshotList is a list of Snapshots.
+type SnapshotList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []Snapshot `json:"items"`
+}
+
+// SnapshotSpec defines the specification for an Ark Snapshot.
+type SnapshotSpec struct {
```

Comment:
take a look at the `VolumeBackupInfo` type in `pkg/apis/ark/v1` to see the fields that we capture today for snapshots.  `VolumeSnapshot` will replace `VolumeBackupInfo` so needs to have at least all of those fields, plus probably a few more related to location.

We also need to decide which belong in `.spec` vs. `.status` -- I don't have an exact answer off the top of my head, but take a stab at it and we can iterate.